lucene/core/src/java/org/apache/lucene/index/MergePolicy.java

```diff
@@ -526,6 +526,19 @@ public abstract MergeSpecification findForcedMerges(
   public abstract MergeSpecification findForcedDeletesMerges(
       SegmentInfos segmentInfos, MergeContext mergeContext) throws IOException;
 
+  /**
+   * Identifies merges that we want to execute (synchronously) on commit. By default, do not synchronously merge on commit.
+   *
+   * If a returned {@link OneMerge} includes a segment already included in a registered merge, then the commit will fail.
+   * Use {@link MergeContext#getMergingSegments()} to determine which segments are currently registered to merge.
+   *
+   * @param segmentInfos the total set of segments in the index (while preparing the commit)
+   * @param mergeContext the IndexWriter to find the merges on
```

Comment:
Hmm, maybe improve this (it's not an `IndexWriter`) to say the merge policy may use this to find already merging segments?